### PR TITLE
JDK-8316741: BasicStroke.createStrokedShape miter-limits failing on small shapes

### PR DESCRIPTION
**Pull request by bourgesl**
For stroked shapes (no Renderer use), the min-join-error check is disabled.
Added new jtreg test to verify join using pixel counts

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316741](https://bugs.openjdk.org/browse/JDK-8316741): BasicStroke.createStrokedShape miter-limits failing on small shapes (**Bug** - P3)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [97d33874](https://git.openjdk.org/jdk/pull/16002/files/97d3387450867c2978c0f2a0075d39e806194b26)
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer) ⚠️ Review applies to [97d33874](https://git.openjdk.org/jdk/pull/16002/files/97d3387450867c2978c0f2a0075d39e806194b26)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16002/head:pull/16002` \
`$ git checkout pull/16002`

Update a local copy of the PR: \
`$ git checkout pull/16002` \
`$ git pull https://git.openjdk.org/jdk.git pull/16002/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16002`

View PR using the GUI difftool: \
`$ git pr show -t 16002`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16002.diff">https://git.openjdk.org/jdk/pull/16002.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16002#issuecomment-1742197949)

### CODE REVIEW

**Comment by bourgesl** [comment]
@prrace could you review this simple fix for jdk22, thatcan be backported to 21 next ?

**Comment by prrace** [comment]
Let me submit a test task to verify.

**Review by prrace — APPROVED** [review]
Tests all passed.

**Comment by bourgesl** [comment]
Thanks @prrace for approval.
@kevinrushforth could you review this Marlin2D patch too as I want to backport it to openjdk 21 asap ?

**Comment by bourgesl** [comment]
@johanvos could you review this fix?

**Review by prrace — APPROVED** [review]
I'm not seeing that github shows I approved this but I certainly intended to. So hitting approve now.
Oh, maybe its because you have a mismatch in the title.

**Comment by bourgesl** [comment]
As only @prrace approved, is it enough for me to integrate this PR or I need another reviewer ?

File: test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+/**
+ * @test
+ * @bug 8316741
+ * @summary Verifies that Marlin renderer's Stroker generates properly joins
+ * in createStrokedShape()
+ * @run main TestCreateStrokedShapeJoins
+ */
+public class TestCreateStrokedShapeJoins {
+
```
Comment (DamonGuy) [inline test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java:39]:
```suggestion

/*
 * @test
 * @bug 8316741
 * @summary Verifies that Marlin renderer's Stroker generates properly joins
 * in createStrokedShape()
 * @run main TestCreateStrokedShapeJoins
 */
 
public class TestCreateStrokedShapeJoins {

```

File: test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    static final boolean SAVE_IMAGE = false;
+
+    private final static int W = 200;
+
+    private final static int[] REF_COUNTS = new int[]{4561, 4790, 5499};
```
Comment (DamonGuy) [inline test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java:44]:
```suggestion
    private final static int[] REF_COUNTS = new int[] {4561, 4790, 5499};
```

File: test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    public static void main(String[] args) throws Exception {
+        final int[] test = new int[]{
```
Comment (DamonGuy) [inline test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java:47]:
```suggestion
        final int[] test = new int[] {
```

**Review by DamonGuy — APPROVED** [review]
Changes look reasonable and the test looks good. Ran it myself and it passes with the changes.

**Comment by DamonGuy** [comment]
> As only @prrace approved, is it enough for me to integrate this PR or I need another reviewer ?

I have additionally approved it to make that decision easier. Typically, two Reviewers is recommended unless it's a really trivial change.

File: test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java (hunk 1 above)
Comment (bourgesl) [reply test/jdk/sun/java2d/marlin/TestCreateStrokedShapeJoins.java:39]: javadoc as usual -> no

**Pull request closed without merging**